pkg/controller/association/ca.go

```diff
@@ -24,40 +22,40 @@ type CASecret struct {
 	CACertProvided bool
 }
 
-// ElasticsearchCACertSecretName returns the name of the secret holding the certificate chain used
-// by the associated resource to establish and validate a secured HTTP connection to Elasticsearch.
-func ElasticsearchCACertSecretName(associated commonv1.Associated, associationName string) string {
+// CACertSecretName returns the name of the secret holding the certificate chain used
+// by the associated resource to establish and validate a secured HTTP connection to the target service.
+func CACertSecretName(associated commonv1.Associated, associationName string) string {
 	return associated.GetName() + "-" + associationName + "-ca"
 }
 
-// ReconcileCASecret keeps in sync a copy of the Elasticsearch CA.
-// It is the responsibility of the controller to set a watch on the ES CA.
-func (r *Reconciler) ReconcileCASecret(associated commonv1.Associated, es types.NamespacedName) (CASecret, error) {
-	publicESHTTPCertificatesNSN := certificates.PublicCertsSecretRef(esv1.ESNamer, es)
+// ReconcileCASecret keeps in sync a copy of the target service CA.
+// It is the responsibility of the association controller to set a watch on this CA.
+func (r *Reconciler) ReconcileCASecret(association commonv1.Association, namer name.Namer, service types.NamespacedName, caSecretServiceName string) (CASecret, error) {
+	servicePublicHTTPCertificatesNSN := certificates.PublicCertsSecretRef(namer, service)
 
-	// retrieve the HTTP certificates from ES namespace
-	var publicESHTTPCertificatesSecret corev1.Secret
-	if err := r.Get(publicESHTTPCertificatesNSN, &publicESHTTPCertificatesSecret); err != nil {
+	// retrieve the HTTP certificates from the service namespace
+	var servicePublicHTTPCertificatesSecret corev1.Secret
+	if err := r.Get(servicePublicHTTPCertificatesNSN, &servicePublicHTTPCertificatesSecret); err != nil {
 		if errors.IsNotFound(err) {
 			return CASecret{}, nil // probably not created yet, we'll be notified to reconcile later
 		}
 		return CASecret{}, err
 	}
 
-	labels := r.AssociationLabels(k8s.ExtractNamespacedName(associated))
+	labels := r.AssociationLabels(k8s.ExtractNamespacedName(association))
 	// Add the Elasticsearch name, this is only intended to help the user to filter on these resources
```

Comment:
This is not the Elasticsearch name in case of the Kibana association but the Kibana name